lib/AST/RawComment.cpp

```diff
@@ -238,3 +239,27 @@ CharSourceRange RawComment::getCharSourceRange() {
                 static_cast<const char *>(Start.getOpaquePointerValue());
   return CharSourceRange(Start, Length);
 }
+
+bool BasicSourceFileInfo::populate(SourceFile *SF) {
+  SourceManager &SM = SF->getASTContext().SourceMgr;
+
+  auto filename = SF->getFilename();
+  if (filename.empty())
+    return true;
+  auto stat = SM.getFileSystem()->status(filename);
+  if (!stat)
+    return true;
+
+  FilePath = filename;
+  LastModified = stat->getLastModificationTime();
+  FileSize = stat->getSize();
+
+  if (SF->hasInterfaceHash()) {
+    InterfaceHash = SF->getInterfaceHashIncludingTypeMembers();
+  } else {
+    // FIXME: Parse the file with EnableInterfaceHash option.
```

Comment:
When would this happen?

Reply:
Whole module optimization. Interface hash is generated only when the file is explicitly specified `-primary-file`.